adapters/repos/db/vector/flat/index.go

```diff
@@ -558,22 +558,52 @@ func (index *flat) findTopVectorsCached(heap *priorityqueue.Queue[any],
 
 	// since keys are sorted, once key/id get greater than max allowed one
 	// further search can be stopped
-	for ; id < uint64(all) && (allow == nil || id <= allowMax); id++ {
+	for id < uint64(all) && (allow == nil || id <= allowMax) {
 		if allow == nil || allow.Contains(id) {
-			vec, err := index.bqCache.Get(context.Background(), id)
-			if err != nil {
-				return err
-			}
-			if len(vec) == 0 {
-				continue
-			}
-			distance, err := index.bq.DistanceBetweenCompressedVectors(vec, vectorBQ)
-			if err != nil {
-				return err
+
+			vecs, errs, start, end := index.bqCache.GetAllInCurrentLock(context.Background(), id)
+
+			for i, vec := range vecs {
```

Comment:
@ajit283 one comment here there may be a possibility of someone having an extremely large tenant they are using the flat index on. Right now if there is say a 20M sized flat tenant we don't do anything to prevent a slow query happening which we would block writes. Should we reacquire locks every 1-2M objects to prevent this?